velox/common/memory/MemoryPool.h

```diff
@@ -241,6 +241,8 @@ class MemoryPool : public std::enable_shared_from_this<MemoryPool> {
       int64_t size,
       std::optional<uint32_t> alignment = std::nullopt) = 0;
 
+  virtual void reportAllocation(int64_t size) = 0;
```

Comment:
Can you add a comment explaining usage for these method as its external by nature ?